04/index.js

```diff
@@ -1,29 +1,24 @@
-
-const ul = document.getElementById('js-lists');
+const ul = document.getElementById("js-lists");
 const fragment = document.createDocumentFragment();
 
-const contents = [
-  { href: 'a1.html', src: '/img/bookmark.png', text: 'a1' },
-  { href: 'a2.html', src: '/img/message.png', text: 'a2' }
+const contents =   [
+    { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+    { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" },
 ];
 
-for(let i = 0; i < contents.length; i++){
-  const image = document.createElement('img');
-  const anchor = document.createElement('a');
-  const li = document.createElement('li');
-
-  const content = contents[i];
-
-  image.src = content.src;
-  anchor.href = content.href;
-
-  li
-  .appendChild(anchor)
-  .appendChild(image);
+contents.forEach ((content)=> {
+  let image = document.createElement('img');
+  let anchor = document.createElement('a');
+  let li = document.createElement('li');
 
-  anchor.insertAdjacentHTML('beforeend',content.text);
+  image.src = content.img;
+  image.alt = content.alt;
+  anchor.href = "/" + content.to;
```

Comment:
You can also write like this.
  anchor.href = `./${content.to}`;
This time, I think it should be ”./”.

Reply:
I changed the notation to use template literals.
I'd like to use this when combining strings and variables as in this case.

ba5b056c97054d8f8a54fc3e1a270ac30645fe37